### ISSUE TITLE
bevdet_test

### ISSUE THREAD

**Issue by wuhueee** [zzj403/BEV_Robust#4]
作者你好，非常感谢你的项目，请问你使用的模型文件是bevdet-r50.pth吗，对应的配置文件是bevdet-r50-adv.py吗，我在进行测试的时候出现了模型权重不一致的问题，可以帮我解答一下吗谢谢！！其中我对你的项目稍微进行了一点修改用于迁移攻击，除了该模型评估结果有问题，其他的都正常。
(bevdet-adv) dengzhijie2@dengzhijie2VM:/nfs/code/BEV_Robust/BEVDet$ CUDA_VISIBLE_DEVICES=0 python tools/test_transfer_attack.py   configs/bevdet/bevdet-r50-adv.
py   ckpts/bevdet-r50.pth   /nfs/code/BEV_Robust/TransFusion/transfer_attack/transfusion_source_eps8_step10/adv_meta   --work-dir transfer_attack/baseline_bevde
t_fixed
Building dataset...
Dataset size: 6019 samples
Building model...
/home/dengzhijie2/anaconda3/envs/bevdet-adv/lib/python3.8/site-packages/mmdet/models/backbones/resnet.py:400: UserWarning: DeprecationWarning: pretrained is deprecated, please use "init_cfg" instead
  warnings.warn('DeprecationWarning: pretrained is deprecated, '
Use load_from_local loader
The model and loaded state dict do not match exactly

size mismatch for pts_bbox_head.task_heads.0.heatmap.1.weight: copying a param with shape torch.Size([10, 64, 3, 3]) from checkpoint, the shape in current model is torch.Size([1, 64, 3, 3]).
size mismatch for pts_bbox_head.task_heads.0.heatmap.1.bias: copying a param with shape torch.Size([10]) from checkpoint, the shape in current model is torch.Size([1]).
size mismatch for img_neck.lateral_convs.0.conv.weight: copying a param with shape torch.Size([256, 1024, 1, 1]) from checkpoint, the shape in current model is torch.Size([512, 1024, 1, 1]).
size mismatch for img_neck.lateral_convs.0.conv.bias: copying a param with shape torch.Size([256]) from checkpoint, the shape in current model is torch.Size([512]).
size mismatch for img_neck.lateral_convs.1.conv.weight: copying a param with shape torch.Size([256, 2048, 1, 1]) from checkpoint, the shape in current model is torch.Size([512, 2048, 1, 1]).
size mismatch for img_neck.lateral_convs.1.conv.bias: copying a param with shape torch.Size([256]) from checkpoint, the shape in current model is torch.Size([512]).
size mismatch for img_neck.fpn_convs.0.conv.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([512, 512, 3, 3]).
size mismatch for img_neck.fpn_convs.0.conv.bias: copying a param with shape torch.Size([256]) from checkpoint, the shape in current model is torch.Size([512]).
unexpected key in source state_dict: img_view_transformer.depth_net.weight, img_view_transformer.depth_net.bias

missing keys in source state_dict: pts_bbox_head.task_heads.1.reg.0.conv.weight, pts_bbox_head.task_heads.1.reg.0.bn.weight, pts_bbox_head.task_heads.1.reg.0.bn.bias, pts_bbox_head.task_heads.1.reg.0.bn.running_mean, pts_bbox_head.task_heads.1.reg.0.bn.running_var, pts_bbox_head.task_heads.1.reg.1.weight, pts_bbox_head.task_heads.1.reg.1.bias, pts_bbox_head.task_heads.1.height.0.conv.weight, pts_bbox_head.task_heads.1.height.0.bn.weight, pts_bbox_head.task_heads.1.height.0.bn.bias, pts_bbox_head.task_heads.1.height.0.bn.running_mean, pts_bbox_head.task_heads.1.height.0.bn.running_var, pts_bbox_head.task_heads.1.height.1.weight, pts_bbox_head.task_heads.1.height.1.bias, pts_bbox_head.task_heads.1.dim.0.conv.weight, pts_bbox_head.task_heads.1.dim.0.bn.weight, pts_bbox_head.task_heads.1.dim.0.bn.bias, pts_bbox_head.task_heads.1.dim.0.bn.running_mean, pts_bbox_head.task_heads.1.dim.0.bn.running_var, pts_bbox_head.task_heads.1.dim.1.weight, pts_bbox_head.task_heads.1.dim.1.bias, pts_bbox_head.task_heads.1.rot.0.conv.weight, pts_bbox_head.task_heads.1.rot.0.bn.weight, pts_bbox_head.task_heads.1.rot.0.bn.bias, pts_bbox_head.task_heads.1.rot.0.bn.running_mean, pts_bbox_head.task_heads.1.rot.0.bn.running_var, pts_bbox_head.task_heads.1.rot.1.weight, pts_bbox_head.task_heads.1.rot.1.bias, pts_bbox_head.task_heads.1.vel.0.conv.weight, pts_bbox_head.task_heads.1.vel.0.bn.weight, pts_bbox_head.task_heads.1.vel.0.bn.bias, pts_bbox_head.task_heads.1.vel.0.bn.running_mean, pts_bbox_head.task_heads.1.vel.0.bn.running_var, pts_bbox_head.task_heads.1.vel.1.weight, pts_bbox_head.task_heads.1.vel.1.bias, pts_bbox_head.task_heads.1.heatmap.0.conv.weight, pts_bbox_head.task_heads.1.heatmap.0.bn.weight, pts_bbox_head.task_heads.1.heatmap.0.bn.bias, pts_bbox_head.task_heads.1.heatmap.0.bn.running_mean, pts_bbox_head.task_heads.1.heatmap.0.bn.running_var, pts_bbox_head.task_heads.1.heatmap.1.weight, pts_bbox_head.task_heads.1.heatmap.1.bias, pts_bbox_head.task_heads.2.reg.0.conv.weight, pts_bbox_head.task_heads.2.reg.0.bn.weight, pts_bbox_head.task_heads.2.reg.0.bn.bias, pts_bbox_head.task_heads.2.reg.0.bn.running_mean, pts_bbox_head.task_heads.2.reg.0.bn.running_var, pts_bbox_head.task_heads.2.reg.1.weight, pts_bbox_head.task_heads.2.reg.1.bias, pts_bbox_head.task_heads.2.height.0.conv.weight, pts_bbox_head.task_heads.2.height.0.bn.weight, pts_bbox_head.task_heads.2.height.0.bn.bias, pts_bbox_head.task_heads.2.height.0.bn.running_mean, pts_bbox_head.task_heads.2.height.0.bn.running_var, pts_bbox_head.task_heads.2.height.1.weight, pts_bbox_head.task_heads.2.height.1.bias, pts_bbox_head.task_heads.2.dim.0.conv.weight, pts_bbox_head.task_heads.2.dim.0.bn.weight, pts_bbox_head.task_heads.2.dim.0.bn.bias, pts_bbox_head.task_heads.2.dim.0.bn.running_mean, pts_bbox_head.task_heads.2.dim.0.bn.running_var, pts_bbox_head.task_heads.2.dim.1.weight, pts_bbox_head.task_heads.2.dim.1.bias, pts_bbox_head.task_heads.2.rot.0.conv.weight, pts_bbox_head.task_heads.2.rot.0.bn.weight, pts_bbox_head.task_heads.2.rot.0.bn.bias, pts_bbox_head.task_heads.2.rot.0.bn.running_mean, pts_bbox_head.task_heads.2.rot.0.bn.running_var, pts_bbox_head.task_heads.2.rot.1.weight, pts_bbox_head.task_heads.2.rot.1.bias, pts_bbox_head.task_heads.2.vel.0.conv.weight, pts_bbox_head.task_heads.2.vel.0.bn.weight, pts_bbox_head.task_heads.2.vel.0.bn.bias, pts_bbox_head.task_heads.2.vel.0.bn.running_mean, pts_bbox_head.task_heads.2.vel.0.bn.running_var, pts_bbox_head.task_heads.2.vel.1.weight, pts_bbox_head.task_heads.2.vel.1.bias, pts_bbox_head.task_heads.2.heatmap.0.conv.weight, pts_bbox_head.task_heads.2.heatmap.0.bn.weight, pts_bbox_head.task_heads.2.heatmap.0.bn.bias, pts_bbox_head.task_heads.2.heatmap.0.bn.running_mean, pts_bbox_head.task_heads.2.heatmap.0.bn.running_var, pts_bbox_head.task_heads.2.heatmap.1.weight, pts_bbox_head.task_heads.2.heatmap.1.bias, pts_bbox_head.task_heads.3.reg.0.conv.weight, pts_bbox_head.task_heads.3.reg.0.bn.weight, pts_bbox_head.task_heads.3.reg.0.bn.bias, pts_bbox_head.task_heads.3.reg.0.bn.running_mean, pts_bbox_head.task_heads.3.reg.0.bn.running_var, pts_bbox_head.task_heads.3.reg.1.weight, pts_bbox_head.task_heads.3.reg.1.bias, pts_bbox_head.task_heads.3.height.0.conv.weight, pts_bbox_head.task_heads.3.height.0.bn.weight, pts_bbox_head.task_heads.3.height.0.bn.bias, pts_bbox_head.task_heads.3.height.0.bn.running_mean, pts_bbox_head.task_heads.3.height.0.bn.running_var, pts_bbox_head.task_heads.3.height.1.weight, pts_bbox_head.task_heads.3.height.1.bias, pts_bbox_head.task_heads.3.dim.0.conv.weight, pts_bbox_head.task_heads.3.dim.0.bn.weight, pts_bbox_head.task_heads.3.dim.0.bn.bias, pts_bbox_head.task_heads.3.dim.0.bn.running_mean, pts_bbox_head.task_heads.3.dim.0.bn.running_var, pts_bbox_head.task_heads.3.dim.1.weight, pts_bbox_head.task_heads.3.dim.1.bias, pts_bbox_head.task_heads.3.rot.0.conv.weight, pts_bbox_head.task_heads.3.rot.0.bn.weight, pts_bbox_head.task_heads.3.rot.0.bn.bias, pts_bbox_head.task_heads.3.rot.0.bn.running_mean, pts_bbox_head.task_heads.3.rot.0.bn.running_var, pts_bbox_head.task_heads.3.rot.1.weight, pts_bbox_head.task_heads.3.rot.1.bias, pts_bbox_head.task_heads.3.vel.0.conv.weight, pts_bbox_head.task_heads.3.vel.0.bn.weight, pts_bbox_head.task_heads.3.vel.0.bn.bias, pts_bbox_head.task_heads.3.vel.0.bn.running_mean, pts_bbox_head.task_heads.3.vel.0.bn.running_var, pts_bbox_head.task_heads.3.vel.1.weight, pts_bbox_head.task_heads.3.vel.1.bias, pts_bbox_head.task_heads.3.heatmap.0.conv.weight, pts_bbox_head.task_heads.3.heatmap.0.bn.weight, pts_bbox_head.task_heads.3.heatmap.0.bn.bias, pts_bbox_head.task_heads.3.heatmap.0.bn.running_mean, pts_bbox_head.task_heads.3.heatmap.0.bn.running_var, pts_bbox_head.task_heads.3.heatmap.1.weight, pts_bbox_head.task_heads.3.heatmap.1.bias, pts_bbox_head.task_heads.4.reg.0.conv.weight, pts_bbox_head.task_heads.4.reg.0.bn.weight, pts_bbox_head.task_heads.4.reg.0.bn.bias, pts_bbox_head.task_heads.4.reg.0.bn.running_mean, pts_bbox_head.task_heads.4.reg.0.bn.running_var, pts_bbox_head.task_heads.4.reg.1.weight, pts_bbox_head.task_heads.4.reg.1.bias, pts_bbox_head.task_heads.4.height.0.conv.weight, pts_bbox_head.task_heads.4.height.0.bn.weight, pts_bbox_head.task_heads.4.height.0.bn.bias, pts_bbox_head.task_heads.4.height.0.bn.running_mean, pts_bbox_head.task_heads.4.height.0.bn.running_var, pts_bbox_head.task_heads.4.height.1.weight, pts_bbox_head.task_heads.4.height.1.bias, pts_bbox_head.task_heads.4.dim.0.conv.weight, pts_bbox_head.task_heads.4.dim.0.bn.weight, pts_bbox_head.task_heads.4.dim.0.bn.bias, pts_bbox_head.task_heads.4.dim.0.bn.running_mean, pts_bbox_head.task_heads.4.dim.0.bn.running_var, pts_bbox_head.task_heads.4.dim.1.weight, pts_bbox_head.task_heads.4.dim.1.bias, pts_bbox_head.task_heads.4.rot.0.conv.weight, pts_bbox_head.task_heads.4.rot.0.bn.weight, pts_bbox_head.task_heads.4.rot.0.bn.bias, pts_bbox_head.task_heads.4.rot.0.bn.running_mean, pts_bbox_head.task_heads.4.rot.0.bn.running_var, pts_bbox_head.task_heads.4.rot.1.weight, pts_bbox_head.task_heads.4.rot.1.bias, pts_bbox_head.task_heads.4.vel.0.conv.weight, pts_bbox_head.task_heads.4.vel.0.bn.weight, pts_bbox_head.task_heads.4.vel.0.bn.bias, pts_bbox_head.task_heads.4.vel.0.bn.running_mean, pts_bbox_head.task_heads.4.vel.0.bn.running_var, pts_bbox_head.task_heads.4.vel.1.weight, pts_bbox_head.task_heads.4.vel.1.bias, pts_bbox_head.task_heads.4.heatmap.0.conv.weight, pts_bbox_head.task_heads.4.heatmap.0.bn.weight, pts_bbox_head.task_heads.4.heatmap.0.bn.bias, pts_bbox_head.task_heads.4.heatmap.0.bn.running_mean, pts_bbox_head.task_heads.4.heatmap.0.bn.running_var, pts_bbox_head.task_heads.4.heatmap.1.weight, pts_bbox_head.task_heads.4.heatmap.1.bias, pts_bbox_head.task_heads.5.reg.0.conv.weight, pts_bbox_head.task_heads.5.reg.0.bn.weight, pts_bbox_head.task_heads.5.reg.0.bn.bias, pts_bbox_head.task_heads.5.reg.0.bn.running_mean, pts_bbox_head.task_heads.5.reg.0.bn.running_var, pts_bbox_head.task_heads.5.reg.1.weight, pts_bbox_head.task_heads.5.reg.1.bias, pts_bbox_head.task_heads.5.height.0.conv.weight, pts_bbox_head.task_heads.5.height.0.bn.weight, pts_bbox_head.task_heads.5.height.0.bn.bias, pts_bbox_head.task_heads.5.height.0.bn.running_mean, pts_bbox_head.task_heads.5.height.0.bn.running_var, pts_bbox_head.task_heads.5.height.1.weight, pts_bbox_head.task_heads.5.height.1.bias, pts_bbox_head.task_heads.5.dim.0.conv.weight, pts_bbox_head.task_heads.5.dim.0.bn.weight, pts_bbox_head.task_heads.5.dim.0.bn.bias, pts_bbox_head.task_heads.5.dim.0.bn.running_mean, pts_bbox_head.task_heads.5.dim.0.bn.running_var, pts_bbox_head.task_heads.5.dim.1.weight, pts_bbox_head.task_heads.5.dim.1.bias, pts_bbox_head.task_heads.5.rot.0.conv.weight, pts_bbox_head.task_heads.5.rot.0.bn.weight, pts_bbox_head.task_heads.5.rot.0.bn.bias, pts_bbox_head.task_heads.5.rot.0.bn.running_mean, pts_bbox_head.task_heads.5.rot.0.bn.running_var, pts_bbox_head.task_heads.5.rot.1.weight, pts_bbox_head.task_heads.5.rot.1.bias, pts_bbox_head.task_heads.5.vel.0.conv.weight, pts_bbox_head.task_heads.5.vel.0.bn.weight, pts_bbox_head.task_heads.5.vel.0.bn.bias, pts_bbox_head.task_heads.5.vel.0.bn.running_mean, pts_bbox_head.task_heads.5.vel.0.bn.running_var, pts_bbox_head.task_heads.5.vel.1.weight, pts_bbox_head.task_heads.5.vel.1.bias, pts_bbox_head.task_heads.5.heatmap.0.conv.weight, pts_bbox_head.task_heads.5.heatmap.0.bn.weight, pts_bbox_head.task_heads.5.heatmap.0.bn.bias, pts_bbox_head.task_heads.5.heatmap.0.bn.running_mean, pts_bbox_head.task_heads.5.heatmap.0.bn.running_var, pts_bbox_head.task_heads.5.heatmap.1.weight, pts_bbox_head.task_heads.5.heatmap.1.bias, img_view_transformer.dx, img_view_transformer.bx, img_view_transformer.nx, img_view_transformer.frustum, img_view_transformer.depthnet.weight, img_view_transformer.depthnet.bias

**Comment by zzj403** [comment]
这很可能是由于我们当初下载的是老版本的bevdet-r50.pth，而你这边可能是新的bevdet-r50.pth，导致的冲突。而打开老版本的 BEVDet 库（从我的repo里），可以发现老版本的weight的分享链接已经失效了。也许你可以问一下BEVDet 库的原作者是否愿意分享老版本（[BEVDet @ 2919b99](https://github.com/HuangJunJie2017/BEVDet/tree/2919b997c8fb2cb4737fd844200936ef04900d37)）的 bevdet-r50.pth 权重。

This is likely because we downloaded the old version of bevdet-r50.pth, while you may have a newer version, causing a conflict. If you open the old version of the BEVDet repository (from my repo), you'll find that the link to share the old weights is no longer valid. Perhaps you could ask the original author of the BEVDet library if they would be willing to share the old version of bevdet-r50.pth ([BEVDet @ 2919b99](https://github.com/HuangJunJie2017/BEVDet/tree/2919b997c8fb2cb4737fd844200936ef04900d37)).

**Comment by wuhueee** [comment]
谢谢作者，你的建议很有帮助！！！愿你以后工作顺利！！！

**Comment by wuhueee** [comment]
您好，老版本的bevdet-r50.pth你有保存吗，可以提供一下吗

**Comment by zzj403** [comment]
> 您好，老版本的bevdet-r50.pth你有保存吗，可以提供一下吗

你可以留一个邮箱，以便通过非公开方式分享给你。
You can leave an email address so that it can be shared with you privately.

**Comment by wuhueee** [comment]
> > 您好，老版本的bevdet-r50.pth你有保存吗，可以提供一下吗
> 
> 你可以留一个邮箱，以便通过非公开方式分享给你。 You can leave an email address so that it can be shared with you privately.

[2480759489@qq.com](mailto:2480759489@qq.com)感谢作者的分享

**Comment by wuhueee** [comment]
作者打扰一下，请问你的显卡的显存容量是多大呢，24g是不是会有点不够
